test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -256,6 +256,55 @@ main() {
               reason: 'should preserve existing inheritance');
         });
 
+        test('with static PropsMeta declaration', () {
```

Comment:
we should also have a test which verifies this functionality on props/state mixins:

```
test('with static PropsMeta declaration in PropsMixin', () {
  final transformedLine = 'static const PropsMeta meta = \$Props(FooPropsMixin);';
  setUpAndGenerate('''
    @PropsMixin()
    class FooPropsMixin {
      static const PropsMeta meta = \$metaForFooPropsMixin;
    }
    '''
  );
  var transformedSource = transformedFile.getTransformedText();
  expect(transformedSource, contains(transformedLine));
});